pycqed/measurement/quantum_experiment.py

```diff
@@ -661,5 +662,74 @@ def save_timers(self, quantum_experiment=True, sequence=True, segments=True, fil
+        :param segments (int, list): index of segments to plot. Plots all
+        segments by default.
+        :param qubits (list): list of qubits to plot
+        :param save (bool): save the figures.
+        :param plot_kwargs: kwargs passed on to segment.plot()
+        :return:
+        """
+        plot_kwargs = deepcopy(plot_kwargs)
+        if sequences is None:
+            # plot all sequences
+            sequences = self.sequences
+        elif isinstance(sequences, int):
+            sequences = [self.sequences[sequences]]
+        elif isinstance(sequences, list):
+            if len(sequences) and isinstance(sequences[0], int):
+                # sequences is a list of indices of sequences to plot
+                sequences = [self.sequences[s] for s in sequences]
+            elif len(sequences) and isinstance(sequences[0], str):
+                # sequences is a list of names of sequences to plot
+                sequences = [[s for s in self.sequences if s.name == seqname]
+                             for seqname in  sequences]
+                sequences = np.ravel(sequences).tolist()
+        if qubits is None:
+            qubits = self.meas_objs
+        plot_kwargs.update(dict(channel_map=plot_kwargs.pop('channel_map',
+                           general.get_channel_map(qubits))))
+        plot_kwargs.update(dict(legend=plot_kwargs.pop('legend',False)))
+
+        if segments is None:
+            # plot all segments
+            segments = [np.arange(len(seq.segments)) for seq in sequences]
+        elif isinstance(segments, int):
+            # single segment from index
+            segments = [[segments] for _ in sequences]
+
+        for seq, segs in zip(sequences, segments):
+            for s in segs:
+                if isinstance(s, int):
+                    s = list(seq.segments.keys())[s]
+                if save:
+                    try:
+                        # FIXME: should we load a_tools in qexp? it means each
+                        #  time we reload qexp, we would have to reset the
+                        #  datafolder... For now, current implementation below
+                        #  will work if a_tools is already loaded.
+                        folder = a_tools.get_folder(timestamp=self.timestamp)
+                    except:
+                        log.warning('Could not determine folder of current '
+                                    'experiment. Sequence plot will be saved in '
+                                    'current directory.')
+                        folder = ""
+                    save_kwargs = dict(fname= folder + "/" +
+                                              "_".join((seq.name, s)) + ".png",
+                                       bbox_inches="tight")
+                    plot_kwargs.update(dict(save_kwargs=save_kwargs,
+                                            savefig=True))
+                seq.segments[s].plot(**plot_kwargs)
```

Comment:
You should collect the return values (if any, i.e., if the user sets show_and_close=False). Either return them in a list or store them as a property?

Reply:
I think at least returning them makes sense. I added that. With respect of making them a property, it is likely not essential but I could see the benefit of doing it. From an object oriented perspective, I think it would make more sense to attach it to the segment (e.g. seg.fig) rather than the QExp. What do you think?